src/context_store/ingestion/pipeline.py

```
@@ -169,114 +209,77 @@ async def _process_chunk(
-            else:
-                # 自分が処理を担当する。
-                # 同一 memo_key の後続リクエストのために、自分自身を Task として登録して実行する。
-                # ここでは直接 asyncio.current_task() を登録したいが、
-                # まだ走っているので、ラッパーを Task 化して登録する。
-                current_task = asyncio.create_task(
-                    self._process_chunk_locked_with_cleanup(
+        # 1. ロックを取得して同一 memo_key のタスクをアトミックにチェック・登録する
+        async with self._locks_mutex:
+            target_task = self._content_results.get(memo_key)
+            if target_task is None:
+                # 2. 自分が処理を担当する（Task として登録して実行）
```

Ruff (0.15.7)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)